### ISSUE TITLE
[java] UnusedPrivateField - false negative with anonymous class

### ISSUE THREAD

**Issue by Jane-PL-256** [pmd/pmd#3626]
**Affects PMD Version:6.39.0** 

**Rule:UnusedPrivateMethod**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatefield>

**Description:**
I think PMD should have reported a warning at line 3, but it doesn't. Hence, I think this is a false negative.

**Code Sample demonstrating the issue:**

```java
public class Tester {
    Object anonymousClass = new Object() {
        private int x = 2021; // Unused, should report a warning
    };
}
```

**Expected outcome:A warning**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
